From 30382b98273e2b22b1d5c7b11008b667fca55c01 Mon Sep 17 00:00:00 2001
From: Andrey Epifanov <aepifanov@mirantis.com>
Date: Fri, 16 Jan 2015 16:29:40 +0300
Subject: [PATCH 1/3] Fix Pluto pidfile deleting

Basically it is workaround, Pluto daemon doesn't delete his pid file
on CentOS while owner of parent dir is not root. It looks like some
tricky nuance with unlink function in pluto or specific  CentOS
behavoiur.

Change-Id: Ie1504ed5e73b7842e37fef384f16ed13a7f93a23
Closes-Bug: 1331502

This is a commit not merged by upstream, see the links below for
reference.

https://bugs.launchpad.net/neutron/+bug/1331502
https://review.openstack.org/#/c/147852/

Signed-off-by: huntxu <mhuntxu@gmail.com>
---
 neutron/services/vpn/device_drivers/ipsec.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/neutron/services/vpn/device_drivers/ipsec.py b/neutron/services/vpn/device_drivers/ipsec.py
index c19b61e..9330a4a 100644
--- a/neutron/services/vpn/device_drivers/ipsec.py
+++ b/neutron/services/vpn/device_drivers/ipsec.py
@@ -304,7 +304,7 @@ class OpenSwanProcess(BaseSwanProcess):
         self.config_file = os.path.join(
             self.etc_dir, 'ipsec.conf')
         self.pid_path = os.path.join(
-            self.config_dir, 'var', 'run', 'pluto')
+            self.config_dir, 'var', 'run', 'pluto') + os.sep
 
     def _execute(self, cmd, check_exit_code=True):
         """Execute command on namespace."""
-- 
2.4.0

